instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpCommonAttributesExtractor.java

```diff
@@ -63,19 +63,21 @@ public void onEnd(
       @Nullable RESPONSE response,
       @Nullable Throwable error) {
 
-    internalSet(
-        attributes, SemanticAttributes.HTTP_REQUEST_CONTENT_LENGTH, requestContentLength(request));
+    Long requestLength = requestContentLength(request);
+    if (requestLength != null && requestLength > 0) {
```

Comment:
isn't requestLength `0` valid (and different from not being captured)?

Reply:
can we apply that change just to the reactor netty HttpAttributesGetter?

Reply:
I don't think we can (well, not in a "nice" way) - since the getter simply returns the headers, which then get parsed in the extractor.
On a second thought, maybe we can live with this: the `content-length=0` header is actually set on the outgoing HTTP request, so it's not like the instrumentation is lying. It's just a bit strange to see that in a GET request.